addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/AutomationPanel.java

```diff
@@ -0,0 +1,354 @@
+            planScrollpane.setViewportView(tree);
+        }
+        return planScrollpane;
+    }
+
+    private JScrollPane getOutputScrollpane() {
+        if (outputScrollpane == null) {
+            outputScrollpane = new JScrollPane();
+            outputScrollpane.setViewportView(this.getOutputArea());
+        }
+        return outputScrollpane;
+    }
+
+    private JTextPane getOutputArea() {
+        if (outputArea == null) {
+            outputArea = new JTextPane();
+            outputArea.setEditable(false);
+            styleError = this.getOutputArea().addStyle("Error", null);
+            StyleConstants.setIcon(styleError, RED_BALL_ICON);
+
+            styleWarning = this.getOutputArea().addStyle("Warning", null);
+            StyleConstants.setIcon(styleWarning, ORANGE_BALL_ICON);
+
+            styleInfo = this.getOutputArea().addStyle("Info", null);
+            StyleConstants.setIcon(styleInfo, WHITE_BALL_ICON);
+        }
+        return outputArea;
+    }
+
+    private PlanTreeTableModel getTreeModel() {
+        if (treeModel == null) {
+            treeModel = new PlanTreeTableModel(new DefaultMutableTreeNode("Plan"));
+        }
+        return treeModel;
+    }
+
+    private static String listToStr(List<String> list) {
+        return String.join("\n", list);
+    }
+
+    private void outputMessage(String message, Style style) {
+        StyledDocument doc = this.getOutputArea().getStyledDocument();
+        try {
+            doc.insertString(doc.getLength(), " ", style);
+            doc.insertString(doc.getLength(), "  " + message + "\n", null);
+        } catch (BadLocationException e) {
+            // Ignore
+        }
+    }
+
+    private AutomationPlan getPlan(Event event) {
+        String planIdStr = event.getParameters().get(AutomationEventPublisher.PLAN_ID);
+        if (planIdStr == null) {
+            return null;
+        }
+        int planId;
+        try {
+            planId = Integer.parseInt(planIdStr);
+        } catch (NumberFormatException e) {
+            return null;
+        }
+        return ext.getPlan(planId);
+    }
+
+    private AutomationJob getJob(Event event) {
+        AutomationPlan plan = this.getPlan(event);
+        if (plan == null) {
+            return null;
+        }
+        String jobIdStr = event.getParameters().get(AutomationEventPublisher.JOB_ID);
+        if (jobIdStr == null) {
+            return null;
+        }
+        int jobId;
+        try {
+            jobId = Integer.parseInt(jobIdStr);
+        } catch (NumberFormatException e) {
+            return null;
+        }
+        return plan.getJob(jobId);
+    }
+
+    private void updateJob(Event event) {
+        AutomationJob job = this.getJob(event);
+        if (job != null) {
+            getTreeModel().jobChanged(job);
+        }
+    }
+
+    @Override
+    public void eventReceived(Event event) {
+        switch (event.getEventType()) {
```

Comment:
This should be done in the EDT, since the plan/events are running in another thread.